crates/db_schema/src/impls/tag.rs

```diff
@@ -22,6 +24,66 @@ impl Tag {
       .await
       .with_lemmy_type(LemmyErrorType::NotFound)
   }
+
+  pub async fn community_override_all_from_apub(
+    pool: &mut DbPool<'_>,
+    community_id: CommunityId,
+    ap_tags: Vec<TagInsertForm>,
+  ) -> LemmyResult<()> {
```

Comment:
Instead of all this below, you could just do an upsert, which we have a lot of over the codebase.

https://github.com/LemmyNet/lemmy/blob/main/crates/db_schema/src/impls/person.rs#L74

Reply:
Not DB deletes, but here its just the `deleted` column, which makes it an update, so upsert should work.